cmd/kubelet/app/server.go

```diff
@@ -375,13 +371,6 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies) (err error) {
 			if err != nil {
 				glog.Warningf("Failed to create API Server client for heartbeat: %v", err)
 			}
-		} else {
-			switch {
-			case s.RequireKubeConfig:
-				return fmt.Errorf("invalid kubeconfig: %v", err)
```

Comment:
`--kubeconfig=foo` is now equivalent to `--kubeconfig=foo --require-kubeconfig`

if you provide `--kubeconfig` and createAPIServerClientConfig fails, we have to return an error. that means we need to hoist the return error case up to where we create the client config:

```go
clientConfig, err := createAPIServerClientConfig(s)
if err != nil {
  return fmt.Errorf("invalid kubeconfig: %v", err)
}
```

and if we're returning on non-nil error, you can drop the `if err == nil {` check and unindent the rest

Reply:
OK， I prefer to returning err when `createAPIServerClientConfig` fails.